binlog_driver/cmd/mysql/mysql.go

```diff
@@ -0,0 +1,266 @@
+				if err != nil {
+					log.Error(err)
+				}
+
+			}
+			err = tx.Commit()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+		}
+	}
+
+}
+
+func columnToArg(c *pb.Column) (arg interface{}) {
+	if c.GetIsNull() {
+		return nil
+	}
+
+	if c.Int64Value != nil {
+		return c.GetInt64Value()
+	}
+
+	if c.Uint64Value != nil {
+		return c.GetUint64Value()
+	}
+
+	if c.DoubleValue != nil {
+		return c.GetDoubleValue()
+	}
+
+	if c.BytesValue != nil {
+		return c.GetBytesValue()
+	}
+
+	return c.GetStringValue()
+}
+
+func tableToSQL(table *pb.Table) (sqls []string, sqlArgs [][]interface{}) {
+	replace := func(row *pb.Row) {
+		sql := fmt.Sprintf("replace into `%s`.`%s`", table.GetSchemaName(), table.GetTableName())
+
+		var names []string
+		var holder []string
+		for _, c := range table.GetColumnInfo() {
+			names = append(names, c.GetName())
+			holder = append(holder, "?")
+		}
+		sql += "(" + strings.Join(names, ",") + ")"
+		sql += "values(" + strings.Join(holder, ",") + ")"
+
+		var args []interface{}
+		for _, col := range row.GetColumns() {
+			args = append(args, columnToArg(col))
+		}
+
+		sqls = append(sqls, sql)
+		sqlArgs = append(sqlArgs, args)
+	}
+
+	constructWhere := func() (sql string, usePK bool) {
+		var whereColumns []string
+		for _, col := range table.GetColumnInfo() {
+			if col.GetIsPrimaryKey() {
+				whereColumns = append(whereColumns, col.GetName())
+				usePK = true
+			}
+		}
+		// no primary key
+		if len(whereColumns) == 0 {
+			for _, col := range table.GetColumnInfo() {
+				whereColumns = append(whereColumns, col.GetName())
+			}
+		}
+
+		sql = " where "
+		for i, col := range whereColumns {
+			if i != 0 {
+				sql += " and "
+			}
+
+			sql += fmt.Sprintf("%s = ? ", col)
+		}
+
+		sql += " limit 1"
+
+		return
+	}
+
+	for _, mutation := range table.Mutations {
+		switch mutation.GetType() {
+		case pb.MutationType_Insert:
+			replace(mutation.Row)
+		case pb.MutationType_Update:
+			columnInfo := table.GetColumnInfo()
+			sql := fmt.Sprintf("update `%s`.`%s` set ", table.GetSchemaName(), table.GetTableName())
+			// construct c1 = ?, c2 = ?...
+			for i, col := range columnInfo {
+				if i != 0 {
+					sql += ","
+				}
+				sql += fmt.Sprintf("%s = ? ", col.Name)
+			}
+
+			where, usePK := constructWhere()
+			sql += where
+
+			row := mutation.Row
+			changedRow := mutation.ChangeRow
+
+			var args []interface{}
+			// for set
+			for _, col := range row.GetColumns() {
+				args = append(args, columnToArg(col))
+			}
+
+			// for where
+			for i, col := range changedRow.GetColumns() {
+				if !usePK || columnInfo[i].GetIsPrimaryKey() {
+					args = append(args, columnToArg(col))
+				}
+			}
+
+			sqls = append(sqls, sql)
+			sqlArgs = append(sqlArgs, args)
+
+		case pb.MutationType_Delete:
+			columnInfo := table.GetColumnInfo()
+			sql := fmt.Sprintf("delete from `%s`.`%s` ", table.GetSchemaName(), table.GetTableName())
+
+			where, usePK := constructWhere()
+			sql += where
```

Comment:
how about write like this:
sql := fmt.Sprintf("delete from `%s`.`%s`  %s", table.GetSchemaName(), table.GetTableName(), where)

Reply:
done